org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacCompilationUnitResolver.java

```diff
@@ -488,20 +487,21 @@ public CompilationUnit toCompilationUnit(org.eclipse.jdt.internal.compiler.env.I
 			WorkingCopyOwner workingCopyOwner, WorkingCopyOwner typeRootWorkingCopyOwner, int flags, IProgressMonitor monitor) {
 
 		// collect working copies
-		var workingCopies = JavaModelManager.getJavaModelManager().getWorkingCopies(workingCopyOwner, true);
-		if (workingCopies == null) {
-			workingCopies = new ICompilationUnit[0];
-		}
+		// do we need this
+//		var workingCopies = JavaModelManager.getJavaModelManager().getWorkingCopies(workingCopyOwner, true);
```

Comment:
The main purpose of this is to make sure the compiler gets the in-memory version of all the open files instead of the on-disk version. Some tests modify the compilation units' content in memory without saving to disk and expect the results to change.

Maybe there is smarter way to handle this, though.

Reply:
You can see how Java LS solves it - https://github.com/eclipse-jdtls/eclipse.jdt.ls/blob/8a47af021fe3b31469368d66b1d01ff940427d07/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/BaseDocumentLifeCycleHandler.java#L281

>  Some tests modify the compilation units' content in memory without saving to disk and expect the results to change

Do you know such a test?

Reply:
> Theres IOpenable.hasUnsavedChanges() that we could use to filter out any working copies without unsaved changes, which might help.

I will try.